### ISSUE TITLE
Refactor code around constants.ProfileName to introduce profile.GetCurrentProfile()

### ISSUE THREAD

**Issue by LalatenduMohanty** [minishift/minishift#1769]
We have constants.ProfileName to represent the current profile being used by the user. 
Example:

Lets assume we have two profiles, one is `minishift` and another is `foo`. `minishift` is the active profile. 
 * when user runs ` $ minishift status --profile foo`,  current profile is  `foo` even if active profile is `minishift`. 
 * If user runs the same command without `--profile` i.e. ` $ minishift status` , current profile is `minishift`.

However I see we tend to use profile.GetActiveProfile() to get the current profile as it feels intuitive somehow. I will suggest we introduce   profile.GetCurrentProfile() and use it in place of constants.ProfileName.

**Comment by gbraad** [comment]
Is this part of the current milestone and which probmlem does it solve?

I am actually confused about the name `profile.GetCurrentProfile()`, as this would mean the **current 'active' profile**... right? When `--profile` is used, it is neither the active, nor the current profile, but rather the *specified* profile. Terminology is unclear....

**Comment by LalatenduMohanty** [comment]
@gbraad We can change the function name to make it more intuitive. may be `GetProfileInUse()` . The reason behind the refactor is that I have seen in multiple PRs that we tend to use profile.GetActiveProfile() instead of constants.ProfileName.

**Comment by gbraad** [comment]
I am all for abstracting this (to prevent use of the constants directly), and the suggested name makes more sense, although on 2nd thought still not really close to the intention, as 'in use' has a different meaning.

**Comment by hferentschik** [comment]
I agree with @gbraad, a function is preferable. The problem is that there is a disconnect here. `profile.GetActiveProfile()` really makes you think it gives you the active profile. Effectively you expect this to be `constants.ProfileName`. However, it is basically returning what's in instance config which is not always the right thing. So one needs a better name for what's currently implemented as `profile.GetActiveProfile()` as well.
